kedro_azureml/datasets/folder_dataset.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from typing import Any, Dict, Optional, Type, Union
+
+from kedro.io.core import (
+    VERSION_KEY,
+    VERSIONED_FLAG_KEY,
+    AbstractDataSet,
+    DataSetError,
+    Version,
+)
+
+from kedro_azureml.datasets.pipeline_dataset import AzureMLPipelineDataSet
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMLFolderDataSet(AzureMLPipelineDataSet):
```

Comment:
What about name of this dataset? `AzureMLAssetDataSet`?

Reply:
We're still experimenting with whether we can implement all 3 asset types in one dataset type or not. We can change this depending on the outcome.